lib/src/utils.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:io/io.dart';
+
+class CommonUtils {
+  ProcessManager manager;
+
+  CommonUtils() {
+    this.manager = new ProcessManager();
+  }
+
+  /// Fetches the configlet file if it doesn't exist already, and returns a
+  /// [Future] with the exit code.
+  Future<int> fetchConfiglet() {
+    File configletFile = new File('bin/configlet');
+
+    if (!configletFile.existsSync()) {
+      print('Fetching configlet...');
+      return Process.run('bin/fetch-configlet', []).then((ProcessResult result) {
+        if (result.exitCode != 0) _showError(exitCode);
+        return result.exitCode;
+      });
+    }
+
+    return new Future.value(0);
+  }
+
+  /// Returns a [Future] with the exit code resulting from running the
+  /// [executable] with [arguments].
+  ///
+  /// If the [executable] isn't executable, returns exit code 1 and shows an
+  /// error message.
+  Future<int> runCmd(String executable, [List<String> arguments = const []]) async {
+    // Commented out because it always return false with "pub".
+    //
+    // final result = isExecutable(executable);
+    // if (result is bool && !result || result is Future && !await result) {
+    //   print('Unable to run "$executable". Make sure that it\'s executable and that you have permission to run it.');
+    //   return new Future.value(1);
+    // }
+
+    Process spawn = await manager.spawn(executable, arguments);
+    if (await spawn.exitCode != 0) _showError(exitCode);
+    return spawn.exitCode;
```

Comment:
If you follow  the below suggestion for `_showError()` method, you can just 
```dart 
return _exitCode(spawn.exitCode));
``` 
and avoid duplicating `if` condition